### PR TITLE
Feature/radius slider filter

### PR DESCRIPTION
**Pull request by arthur-mrgt**
# Add RadiusSlider and Last Known Location Saving Functionality

## Description

This Pull Request introduces two major enhancements to the app: the integration of the `RadiusSlider` component and the ability to save and restore the last known user location. These changes ensure a more personalized user experience and improve the app's functionality during relaunches.

### 1. **Integration of `RadiusSlider`**
- Added the `RadiusSlider` component to allow users to adjust the search radius dynamically.
- Ensured seamless integration with the `FilterScreen` to update the selected radius in real time.
- Incorporated UI tests to validate the functionality and behavior of the `RadiusSlider`.

### 2. **Last Known Location Saving and Restoration**
- Enhanced `LocationViewModel` to save the user's last known location using `AppDataStore`.
- Implemented functionality to restore the last known location when the app is relaunched.
- Updated `MainActivity` and `SignInScreen` to leverage the restored location for profile filtering.
- Ensured proper handling of the restored location in offline mode.

### 3. **UI and Test Enhancements**
- Updated `SignInScreen` to utilize the saved location for fetching filtered profiles, especially in offline scenarios.
- Added tests to validate the integration of `RadiusSlider` and the functionality of saving/restoring the last known location.
- Verified correct UI behavior for `RadiusSlider` and other location-dependent features.

## Testing
- Verified that the `RadiusSlider` adjusts the search radius dynamically and updates the `FilterViewModel` correctly.
- Ensured that the last known location is saved and restored accurately across app restarts.
- All unit tests and UI tests pass successfully.

## Screenshots
Capture d’écran 2024-11-21 184906

## Notes
This PR focuses on improving user experience by enabling dynamic radius adjustments and ensuring continuity of functionality with saved locations across app relaunches.

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/FilterScreenTest.kt [hunk 1]
```diff
@@ -404,6 +407,112 @@ class FilterScreenTest {
...
+
+    // Verify the initial radius value is displayed
+    composeTestRule
+        .onNodeWithTag("RadiusSliderTitle", useUnmergedTree = true) // Updated
+        .assertExists()
+        .assertTextEquals("Search Radius: $initialRadius m")
+
+    // Change the slider value and verify
+    val newRadius = 300
+    val newIndex = listOf(50, 100, 200, 300, 400, 500).indexOf(newRadius)
```
Comment (JanStaszewiczEPFL) [inline app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/FilterScreenTest.kt:499]: To make testing more efficient you can put this list at the top as a const or share it with the FilterScreen to allow compatibility

**Review by JanStaszewiczEPFL — CHANGES_REQUESTED** [review]
Great PR, there are just a few changes that can be made for efficiency.

**Review by JanStaszewiczEPFL — APPROVED** [review]

File: app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/FilterScreenTest.kt (hunk 1 above)
Comment (arthur-mrgt) [reply app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/FilterScreenTest.kt]: I agree !

**Review by Kapestian — APPROVED** [review]
LGTM (on behalf of @JanStaszewiczEPFL)

**Review by Kapestian — APPROVED** [review]
LGTM (on behalf of @JanStaszewiczEPFL)

**Pull request merged**